geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -283,4 +292,77 @@ public String toString() {
   public KnownVersion[] getSerializationVersions() {
     return null;
   }
+
+  private void validateScoreIsDouble(byte[] score) {
+    String scoreString = Coder.bytesToString(score).toLowerCase();
+    switch (scoreString) {
+      case "inf":
+      case "infinity":
+      case "+inf":
+      case "+infinity":
+      case "-inf":
+      case "-infinity":
+        return;
+      default:
+        if (!Coder.bytesToString(score).matches("^[-+]?[0-9]*\\.?[0-9]+([eE][-+]?[0-9]+)?$")) {
+          throw new NumberFormatException(ERROR_NOT_A_VALID_FLOAT);
+        }
+    }
+  }
+
+  static class OrderedSetEntry implements Comparable<OrderedSetEntry> {
+    public byte[] member;
+    public Double score;
+
+    @Override
+    public int compareTo(OrderedSetEntry o) {
+      int comparison = score.compareTo(o.score);
+      if (comparison == 0) {
+        return javaImplementationOfAnsiCMemCmp(o);
+      }
+      return comparison;
+    }
+
+    private int javaImplementationOfAnsiCMemCmp(OrderedSetEntry o) {
+      // Scores equal, try lexical ordering
+      byte[] otherMember = o.member;
+      int shortestLength = Math.min(member.length, otherMember.length);
+      for (int i = 0; i < shortestLength; i++) {
+        int localComp = Byte.toUnsignedInt(member[i]) - Byte.toUnsignedInt(otherMember[i]);
+        if (localComp != 0) {
+          return localComp;
+        }
+      }
+      // shorter array whose items are all equal to the first items of a longer array is
+      // considered 'less than'
+      if (member.length < otherMember.length) {
+        return -1; // member < other
+      } else if (member.length > otherMember.length) {
+        return 1; // other < member
+      }
+      return 0; // totally equal - should never happen, because you can't have two members
+      // with same name...
+    }
+
+    OrderedSetEntry(byte[] member, byte[] score) {
+      this.member = member;
+      this.score = makeDoubleWhileHandlingInfinity(score);
+    }
+
+    private Double makeDoubleWhileHandlingInfinity(byte[] score) {
```

Comment:
I wonder if these conversions belong in the Coder class? I could see having `double Coder.bytesToDouble(byte[])` and `byte[] Coder.doubleToBytes(double)` methods. Lower case primitive doubles please!

Reply:
Sure; Coder would need to handle +- infinity but that kind of implementation detail is best encapsulated in a class like that.

Reply:
`Coder` already has `bytesToDouble()` and `doubleToBytes()` methods, so there's no need to reimplement that here. It's possible that the implementation in `Coder` needs a tweak, but I think Hale may be working on that as part of #6534